pydis_site/apps/home/signals.py

```diff
@@ -0,0 +1,192 @@
+from typing import List, Optional, Type
+
+from allauth.account.signals import user_logged_in
+from allauth.socialaccount.models import SocialAccount, SocialLogin
+from allauth.socialaccount.providers.base import Provider
+from allauth.socialaccount.providers.discord.provider import DiscordProvider
+from allauth.socialaccount.signals import (
+    pre_social_login, social_account_added, social_account_removed,
+    social_account_updated)
+from django.contrib.auth.models import Group, User as DjangoUser
+from django.db.models.signals import post_save, pre_delete, pre_save
+
+from pydis_site.apps.api.models import User as DiscordUser
+from pydis_site.apps.staff.models import RoleMapping
+
+
+class SignalListener:
```

Comment:
This class name is too generic. We might have other signal listener classes other places in the site, so it would probably be good if we make this class name more specific. something like `AllauthSignalListener` would be fine.

Besides, SignalListener sounds a bit like a Django internal, and it feels like it should take arguments that configure which signals to listen to, or something.